### PR TITLE
Improve error message.

### PR DESCRIPTION
**Pull request by bangerth**
Based on today's mailing list post.

/rebuild

### CODE REVIEW

File: source/base/data_out_base.cc [hunk 1]
```diff
@@ -8825,7 +8825,11 @@ DataOutBase::write_hdf5_parallel(
   (void)mesh_filename;
   (void)solution_filename;
   (void)comm;
-  AssertThrow(false, ExcMessage("HDF5 support is disabled."));
+  AssertThrow(false,
```
Comment (peterrum) [inline source/base/data_out_base.cc:8828]: Would it be possible to do it like this: https://github.com/dealii/dealii/blob/6dff4a664823d2a98a46d8ab2e5d4e4ec7a3b439/include/deal.II/base/exceptions.h#L1103-L1107?

**Comment by bangerth** [comment]
I finally got around to updating this.

**Review by peterrum — APPROVED** [review]
Thanks!

**Review by kronbichler — APPROVED** [review]

**Comment by kronbichler** [comment]
The test failure on one machine is unrelated to this PR, so I'm merging.

**Pull request merged**